src/test/java/easytests/core/services/TopicsServiceTest.java

```diff
@@ -3,100 +3,261 @@
-        final TopicEntity topicEntityFirst = Entities.createTopicEntityMock(4, 3, "testfirst");
-        final TopicEntity topicEntitySecond = Entities.createTopicEntityMock(5, 3, "testsecond");
-        final List<TopicEntity> topicsEntities = new ArrayList<>();
-        topicsEntities.add(topicEntityFirst);
-        topicsEntities.add(topicEntitySecond);
+    private List<TopicEntity> getTopicsFixturesEntities() {
+        final List<TopicEntity> topicsEntities = new ArrayList<>(2);
+        topicsEntities.add(this.topicsSupport.getEntityFixtureMock(1));
+        topicsEntities.add(this.topicsSupport.getEntityFixtureMock(2));
         return topicsEntities;
     }
 
-    private List<TopicModelInterface> getTopicsModels() {
+    private List<TopicModelInterface> getTopicsFixturesModels() {
         final List<TopicModelInterface> topicsModels = new ArrayList<>(2);
-        for (TopicEntity topicEntity: this.getTopicsEntities()) {
-            topicsModels.add(this.mapTopicModel(topicEntity));
-        }
+        topicsModels.add(this.topicsSupport.getModelFixtureMock(1));
+        topicsModels.add(this.topicsSupport.getModelFixtureMock(2));
         return topicsModels;
     }
 
+    private void assertServiceSet(TopicsOptionsInterface topicOptions) throws Exception {
+        verify(topicOptions, times(1)).setTopicsService(any(TopicsServiceInterface.class));
+        verify(topicOptions, times(1)).setSubjectsService(any(SubjectsServiceInterface.class));
+        verify(topicOptions, times(1)).setQuestionsService(any(QuestionsServiceInterface.class));
+    }
+
+    @Test
+    public void testFindAllPresentList() throws Exception {
+        final List<TopicEntity> topicsEntities = this.getTopicsFixturesEntities();
+        when(this.topicsMapper.findAll()).thenReturn(topicsEntities);
+
+        final List<TopicModelInterface> topicsFoundedModels = this.topicsService.findAll();
+
+        this.topicsSupport.assertModelsListEquals(topicsFoundedModels, this.getTopicsFixturesModels());
+    }
+
+    @Test
+    public void testFindAllAbsentList() throws Exception {
+        when(this.topicsMapper.findAll()).thenReturn(new ArrayList<>(0));
+
+        final List<TopicModelInterface> topicsFoundedModels = this.topicsService.findAll();
+
+        Assert.assertNotNull(topicsFoundedModels);
+        Assert.assertEquals(0, topicsFoundedModels.size());
+    }
+
     @Test
-    public void testFindBySubject() throws Exception {
-        final Integer subjectId = 3;
-        final SubjectModelInterface subjectModel = Mockito.mock(SubjectModelInterface.class);
-        final List<TopicEntity> topicsEntities = this.getTopicsEntities();
-        final List<TopicModelInterface> topicsModels = getTopicsModels();
+    public void testFindAllWithOptions() throws Exception {
+        final ArgumentCaptor<List> listCaptor = ArgumentCaptor.forClass(List.class);
+        // Why mockito's 'when' crashing if I want to pass 'this.get...Entitites()' as a parameter
```

Comment:
Because nested Mocks are unavailable, so we need prepare mocks before when cases. Remove comment pls.